### PR TITLE
[ENH] add option to include derivatives in BIDSDataGrabber

### LINKED ISSUE #2814: BIDSDataGrabber does not support derivatives domain
### Summary
The BIDSDataGrabber Interface cannot use the entities in the derivatives domain (as far as I can tell)

### Actual behavior
If I point the BIDSDataGrabber Interface at a derivatives directory, none of the files are selected

### Expected behavior
The BIDSDataGrabber interface should optionally use derivatives entities.

### How to replicate the behavior
Point the BIDSDataGrabber interface at an `fmriprep` output directory as the base_dir.

### Script/Workflow details
I patched the BIDSDataGrabber interface by [adding a boolean for the input spec to specify derivatives](https://github.com/HBClab/atlascorr/blob/7a85dde2e38427d51fb5d03492056c26706b9ead/atlascorr/interfaces.py#L9-L10) and I [added some logic and changed the call to the layout command](https://github.com/HBClab/atlascorr/blob/7a85dde2e38427d51fb5d03492056c26706b9ead/atlascorr/interfaces.py#L20-L25) to explicitly state which domains are to be used.

I can create a pull request if this is an actual issue.

### Platform details:

```
{'commit_hash': '9f0ca09',
 'commit_source': 'repository',
 'networkx_version': '2.1',
 'nibabel_version': '2.2.1',
 'nipype_version': '1.1.7-dev+g9f0ca09',
 'numpy_version': '1.14.3',
 'pkg_path': '/media/Data/Documents/devel/nipype/nipype',
 'scipy_version': '1.1.0',
 'sys_executable': '/home/james/.conda/envs/atlascorr/bin/python',
 'sys_platform': 'linux',
 'sys_version': '3.6.7 |Anaconda, Inc.| (default, Oct 23 2018, 19:16:44) \n'
                '[GCC 7.3.0]',
 'traits_version': '4.6.0'}
```

### Execution environment
Choose one
-[x] My python environment outside container

### PR DESCRIPTION
**Pull request by jdkent**
## Summary

The behavior of pybids changed from `0.5.*` to `0.6.*` such that the derivatives domain is not included in the default search pattern.
Fixes #2814

This pull request is partially overlapping with #2737

## List of changes proposed in this PR (pull-request)

- add `domains` option in BIDSDataGrabber for user to select what domains should be used for indexing files
- add checks to modify behavior depending on which version of pybids is being used.
## Acknowledgment

- [x] \(Mandatory\) I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

**Comment by effigies** [comment]
@jdkent Sorry, I haven't had time to really think this through. Especially given that it may conflict with (I haven't checked) @adelavega's #2737 (which also needs to merge master), I think we should put this off to next release.

File: nipype/interfaces/io.py [hunk 1]
```diff
@@ -2808,14 +2813,31 @@ def __init__(self, infields=None, **kwargs):
...
         except ImportError:
             from bids.grabbids import BIDSLayout
+
+        pybids_ver = version.parse(bids.__version__)
         exclude = None
         if self.inputs.strict:
             exclude = ['derivatives/', 'code/', 'sourcedata/']
-        layout = BIDSLayout(self.inputs.base_dir, exclude=exclude)
+
+        if pybids_ver < version.parse('0.5'):
```
Comment (adelavega) [inline nipype/interfaces/io.py]: Maybe you can include something like this in my PR? My PR is  more general update to 0.7, and includes some extra functionality / updates

Comment (adelavega) [reply nipype/interfaces/io.py]: Although now that I think about it,  I wouldn't worry too much about backwards compatibility. I would make 0.7 the hard requirement. The default behavior of my PR is to look for derivatives in `/derivatives`, but optionally other paths can be added.

**Comment by effigies** [comment]
Closing in favor of #2737. If further changes are needed, please open a new PR. (I think trying to merge and salvage a usable commit history from this one is probably not worth it.)

**Pull request closed without merging**